packages/dds/tree/src/simple-tree/schemaFactory.ts

```diff
@@ -131,17 +131,27 @@ type UnbrandedName<T extends FlexLeafNodeSchema> = T["name"] extends TreeNodeSch
  *
  * @sealed @public
  */
-export class SchemaFactory<TScope extends string = string, TName extends number | string = string> {
+export class SchemaFactory<
+	TScope extends string | undefined = string,
+	TName extends number | string = string,
+> {
 	private readonly structuralTypes: Map<string, TreeNodeSchema> = new Map();
 
 	/**
 	 * @param scope - Prefix appended to the identifiers of all {@link TreeNodeSchema} produced by this builder.
 	 * Use of [Reverse domain name notation](https://en.wikipedia.org/wiki/Reverse_domain_name_notation) or a UUIDv4 is recommended to avoid collisions.
+	 * You may opt out of using a scope by passing `undefined`, but note that this increases the risk of collisions.
 	 */
 	public constructor(public readonly scope: TScope) {}
 
-	private scoped<Name extends TName | string>(name: Name): `${TScope}.${Name}` {
-		return `${this.scope}.${name}`;
+	// TODO: is `| string` needed here?
```

Comment:
IIRC its needed do deal with the structurally typed cases which derive strings for their names (ex: an array without an explicit name) which can still be used if the TName type does not allow such names.

Reply:
Ah, right, makes sense.